### DISCUSSION TITLE
Deformation of supports

### DISCUSSION THREAD

**Discussion by MartinSalak** [StructuralAnalysisFormat/StructuralAnalysisFormat-Doc#112] [Ideas]
We would like to show you new SAF objects for imposed deformation of supports. We will keep our current format of loads and create two new objects; one for translation and one for rotation. For now in SAF, supports' axes are in the global coordinate system, so the deformation will be set only in the global directions for now.

We would love to know what do you think about it? Would you prefer to set coordinate system for the load rather than use the coordinate system of the support? Would you like the option to set the deformation in absolute values as well (for example 10 mm below global 0)?
![image](https://user-images.githubusercontent.com/89633938/155330750-7e6404b4-9077-44e7-b917-24df2cc79d88.png)

Object: **StructuralPointDisplacement**
Displacement can be imposed in a direction of one of the local axes of a support in a node (StructuralPointSupport). Positive displacement acts in the positive direction of the corresponding axis.  A rigid support is expected in the direction of the displacement. 
| Property | Example |Description |
| --- | --- | --- |
| Name | TRS1 | Human readable unique name of the force |
| Support in a node | Sn6 |The name of the support in a node on which the deformation load is applied |
| Direction | Z | Specifies the direction of the deformation |
| Value [mm] | -10 | Specifies the size of the deformation in the set direction |
| Load case | LC5 | The name of the load case to which the load belongs to |
| Id |  | Unique attribute designation |

![image](https://user-images.githubusercontent.com/89633938/155330161-97ee3677-7eba-4796-990e-93e714808fba.png)

Object: **StructuralPointRotation**
Rotation can be imposed around one of the local axes of support in a node (StructuralPointSupport). Positive rotation acts right-handed about the corresponding positive axis. A rigid support is expected in the direction of the rotation.
| Property | Example |Description |
| --- | --- | --- |
| Name | RS1 | Human readable unique name of the force |
| Support in a node | Sn6 |The name of the support in a node on which the deformation load is applied |
| Direction | Rx | Specifies the direction of the deformation |
| Value [mrad] | -10 | Specifies the size of the deformation in the set direction |
| Load case | LC5 | The name of the load case to which the load belongs to |
| Id |  | Unique attribute designation |

![image](https://user-images.githubusercontent.com/89633938/155330211-c1fda58b-9ad3-446e-b847-9a043e980248.png)

Let us know what you think!

**Comment by sonkaste17** [comment]
Hi @msalak-scia, let's put these two proposals together in one object.

**Comment by MartinSalak** [comment]
New object is implemented in the latest version of documentation which will be release with the next version of SAF, take a look:
https://www.saf.guide/en/latest/loads/structuralpointsupportdef.html

What do you think? All comments and notes are welcome.
